nautobot_design_builder/design_job.py

```diff
@@ -268,6 +268,8 @@ def _run_in_transaction(self, **kwargs):  # pylint: disable=too-many-branches, t
         This version of `run` is wrapped in a transaction and will roll back database changes
         on error. In general, this method should only be called by the `run` method.
         """
+        sid = transaction.savepoint()
```

Comment:
@abates , I think that is a better approach to rollback all the changes from a Design Job if it fails.
This means that neither a Deployment or a ChangeSet will be created if it fails or it's run in dry run mode.